src/Docker/Manager/ContainerManager.php

```diff
@@ -209,6 +210,66 @@ public function run(Container $container, callable $attachCallback = null, array
         return null;
     }
 
+// XX
+    /**
+     * Execute a command in a running container
+     *
+     * @param \Docker\Container $container
+     *
+     * @throws \Docker\Exception\UnexpectedStatusCodeException
+     *
+     * @return InteractiveStream
+     */
+    public function exec(Container $container)
+    {
+        $body = [
+            'AttachStdin'  => false,
+            'AttachStdout' => true,
+            'AttachStderr' => true,
+            'Tty'          => false,
+            'Cmd' => [ '/bin/date' ]
+            #'Cmd' => [ '/bin/bash', '-c', "ls /var/www/html" ]
+        ];
+        $response = $this->client->post(['/containers/{id}/exec', ['id' => $container->getId()]], [
+            'body'         => Json::encode($body),
+            'headers'      => ['content-type' => 'application/json'],
+            'wait'         => true,
+        ]);
+
+        if ($response->getStatusCode() !== "201") {
+            throw UnexpectedStatusCodeException::fromResponse($response);
+        }
+// why is the body empty at this point? Status code of 203 is fine, and content length is 74
+print_r('Status code=' . $response->getStatusCode(). "\n");
+
+// why does the following fail with:
+// PHP Fatal error:  Call to undefined method GuzzleHttp\Message\Response::getContentLength() 
+// see also http://api.guzzlephp.org/class-Guzzle.Http.Message.Response.html
+//print_r('getContentLength=' . $response->getContentLength(). "\n");
+
+print_r('Response string=' . $response->__toString() . "\n");
+$body = json_decode($response->getBody(true));
```

Comment:
Can you try to get body before changing response to string ?

Guzzle is very annoying as once you fetch the response you cannot call the method after as the stream is already read and so il will return an empty string.

If this is not working try to see what happens in the DockerAdapter